src/tools/annotation/bidirectionalTool/moveHandle/movePerpendicularLine.js

```diff
@@ -0,0 +1,87 @@
+import external from './../../../../externalModules.js';
+import getDistance from './getDistance.js';
+import getLineVector from './getLineVector.js';
+import getBaseData from './getBaseData.js';
+
+function isFixedEnd(baseData) {
```

Comment:
instead of taking all of `baseData` reduce to data required?

Reply:
Done.
Thank you!